### PR TITLE
unit testing code for l2_interfaces module added

### PR DESCRIPTION
**Pull request by kannankvs**
##### SUMMARY
l2_interfaces module Unit Test Code Added

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
l2_interfaces module Unit Testing

### CODE REVIEW

**Comment by kannankvs** [comment]
'replaced' and 'overridden' features are getting modified in l2_interfaces module by developers in parallel. This UT PR can be merged once if the PR for l2_interfaces module for 'replaced' and 'overridden' support is merged. If this UT requires rework after merging the l2_interfaces changes, it will be taken up after that gets merged.

File: plugins/module_utils/network/sonic/argspec/l2_interfaces/l2_interfaces.py
Comment (kerry-meyer) [inline plugins/module_utils/network/sonic/argspec/l2_interfaces/l2_interfaces.py:1]: Please delete this file from the change set. Let's wait until the real "replaced" and "overridden" support is added to do these changes. (This comment applies to any other UT modules where the module and argspec are being updated to add these states before the underlying code support has been provided.)

File: plugins/modules/sonic_l2_interfaces.py
Comment (kerry-meyer) [inline plugins/modules/sonic_l2_interfaces.py:1]: Please delete this file from the change set. Let's wait until the real "replaced" and "overridden" support is added to do these changes. (This comment applies to any other UT modules where the module and argspec are being updated to add these states before the underlying code support has been provided.)

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
Please delete the argspec and module file from the change set. Let's wait until the real "replaced" and "overridden" support is added to do these changes. (This comment applies to any other UT modules where the module and argspec are being updated to add these states before the underlying code support has been provided.)

**Comment by kannankvs** [comment]
Changes related to "replaced" and "overridden" in plugins directory are not required and hence removed them now. They should be synced from PR221

File: plugins/modules/sonic_l2_interfaces.py
Comment (kerry-meyer) [reply plugins/modules/sonic_l2_interfaces.py:1]: Alternatively, this PR will have to wait until after PR https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/221 is merged.

**Comment by kannankvs** [comment]
@kerry-meyer : I removed code for testing replaced and overridden. When I remove UT for replaced and overridden, the coverage percentage decreases below 70. Hence, I added few new combinations and improved the coverage above 70% now. Please review and merge.

File: plugins/modules/sonic_l2_interfaces.py
Comment (kerry-meyer) [reply plugins/modules/sonic_l2_interfaces.py:1]: Thanks for making the additional changes.

**Review by kerry-meyer — APPROVED** [review]

**Pull request merged**
